### PR TITLE
Fix: Add source property for fatal errors (fixes #3325)

### LINKED ISSUE #3325: Return source for fatal errors
Currently, when a parsing (`fatal`) error occurs, the `source` is not included in the message that ESLint returns.  It may be worthwhile to investigate whether the source could be returned, which is used in some formatters to show clearly where the problem is.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25500931-return-source-for-fatal-errors?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by gyandeeps**
@IanVS @nzakas is this what is expected?

### CODE REVIEW

File: lib/eslint.js [hunk 1]
```diff
@@ -553,11 +555,12 @@ module.exports = (function() {
 
             // If the message includes a leading line number, strip it:
             var message = ex.message.replace(/^line \d+:/i, "").trim();
+            var source = (ex.lineNumber) ? text.split("\n")[ex.lineNumber - 1].replace("\r", "") : null;
```
Comment (nzakas) [inline lib/eslint.js]: You need to use the correct regex for splitting lines. See SourceCode.

**Comment by nzakas** [comment]
Yes, this is the intent, but we need to use the ES line terminators, not just `\n`

File: lib/eslint.js (hunk 1 above)
Comment (gyandeeps) [inline lib/eslint.js]: Sounds good. will add a static function on `SourceCode` class which can be used to split the code into lines. Does that sound good @nzakas ?

Comment (nzakas) [inline lib/eslint.js]: :+1: just remember to update the Node.JS API docs with that info.

**Comment by gyandeeps** [comment]
@nzakas if the name of the function sounds good then we are good to go .

**Comment by nzakas** [comment]
Hmmm, not quite right. Maybe `splitLines()`?

**Comment by gyandeeps** [comment]
Sure, will make that change.

**Comment by gyandeeps** [comment]
@nzakas Done.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**